forge/forge/op/loss.py

```diff
@@ -73,3 +73,18 @@ def forward(self, prediction, labels):
             return c_reduce
 
         raise RuntimeError("Unsupported reduce type: " + self.reduction)
+
+
+class NLLLoss(ForgeModule):
+    def __init__(self, name: str, reduction: str = "avg"):
+        super().__init__(name)
+        self.reduction = reduction
+        self.is_loss = True
+
+    def forward(self, prediction, labels):
+        zeroed_prediction = Multiply("mul_pred_labels", prediction, labels)
```

Comment:
I would suggest to use another name here - labels are one-hot in majority of cases, but can be an arbitrary distribution.